NotationV0/src/lib/variables.ts

```diff
@@ -1,22 +1,34 @@
 import * as taskLib from 'azure-pipelines-task-lib/task';
+import { STATUS, WARNING } from './constants';
 
 // get artifact references from input or from previous docker push task
 // through "RESOURCE_URIS" variable.
 export function getArtifactReferences(): string[] {
     let artifactRefs = []
     const artifactRefsInput = taskLib.getInput('artifactRefs', false);
     if (!artifactRefsInput) {
+        console.log(taskLib.loc('TryToGetArtifactRefsFromDockerTask'));
         artifactRefs = getArtifactReferencesFromDockerTask();
-        console.log(taskLib.loc('GotArtifactRefsFromDockerTask'));
     } else {
-        artifactRefs = artifactRefsInput.split(',');
+        artifactRefs = artifactRefsInput.split(',').map((artifactRef) => artifactRef.trim());
     }
 
     if (artifactRefs.length === 0) {
         throw new Error(taskLib.loc('ArtifactRefsNotSpecified'));
     }
+
+    // validate repeated artifact references
+    let uniqueArtifactRefs = new Set<string>();
+    for (const artifactRef of artifactRefs) {
+        if (uniqueArtifactRefs.has(artifactRef)) {
+            taskLib.warning(taskLib.loc('RepeatedArtifactRef', artifactRef));
+            taskLib.setTaskVariable(STATUS, WARNING);
+        }
+        uniqueArtifactRefs.add(artifactRef);
```

Comment:
The logic here can be more elegant? If `artifactRef` is already in the set, we don't need to add it again.

Reply:
Updated.